ras_party/views/pending_survey_view.py

```diff
@@ -0,0 +1,194 @@
+import logging
+import uuid
+
+import structlog
+from flask import Blueprint, request, make_response, jsonify, current_app
+from sqlalchemy.exc import SQLAlchemyError
+from werkzeug.exceptions import BadRequest, NotFound
+
+from ras_party.controllers import pending_survey_controller
+from ras_party.controllers.business_controller import get_business_by_id
+from ras_party.controllers.notify_gateway import NotifyGateway
+from ras_party.controllers.pending_survey_controller import get_pending_survey_by_batch_number, \
+    confirm_pending_survey
+from ras_party.controllers.respondent_controller import get_respondent_by_id, get_respondent_by_email
+from ras_party.controllers.validate import Validator, Exists
+from ras_party.exceptions import RasNotifyError
+from ras_party.support.public_website import PublicWebsite
+from ras_party.views.account_view import auth
+
+pending_survey_view = Blueprint('pending_survey_view', __name__)
+logger = structlog.wrap_logger(logging.getLogger(__name__))
+
+
+@pending_survey_view.before_request
+@auth.login_required
+def before_pending_survey_view():
+    pass
+
+
+@auth.get_password
+def get_pw(username):
+    config_username = current_app.config['SECURITY_USER_NAME']
+    config_password = current_app.config['SECURITY_USER_PASSWORD']
+    if username == config_username:
+        return config_password
+
+
+@pending_survey_view.route('/pending-survey-users-count', methods=['GET'])
+def pending_survey_users():
+    """
+    Get total users count who are already enrolled and pending transfer survey against business id and survey id
+    """
+    business_id = request.args.get('business_id')
+    survey_id = request.args.get('survey_id')
+    is_transfer = request.args.get('is_transfer', False)
+    if business_id and survey_id:
+        # this is just to validate that the business_id exists
+        get_business_by_id(business_id)
+        response = pending_survey_controller.get_users_enrolled_and_pending_survey_against_business_and_survey(
+            business_id, survey_id, is_transfer)
+        return make_response(jsonify(response), 200)
+    else:
+        raise BadRequest('Business id and Survey id is required for this request.')
+
+
+@pending_survey_view.route('/pending-surveys', methods=['POST'])
+def post_pending_surveys():
+    """
+     Creates new records for share survey
```

Comment:
Would this be more useful in the API docs than here? (probably with fake UUIDs/emails rather than just repeating the variable name)

Reply:
its only for devs, I think its ok to be here but also to add it to the API docs